test/e2e/specs/editor/blocks/cover.spec.js

```diff
@@ -0,0 +1,245 @@
+		const dimColor = await backgrounDimLocator.evaluate(
+			( node ) => node.style.backgroundColor
+		);
+
+		expect( pickedColor ).toEqual( dimColor );
+	} );
+
+	test( 'can set background image using image upload on block placeholder', async ( {
+		page,
+		editor,
+	} ) => {
+		await editor.insertBlock( { name: 'core/cover' } );
+
+		// Create a test image.
+		const testImage = await getTestImage();
+
+		// Create the block using uploaded image.
+		await page
+			.locator( `.wp-block-cover input[type="file"]` )
+			.setInputFiles( testImage );
+
+		const testImageFilename = path.basename( testImage );
+
+		const backgroundImageLocator = page.locator(
+			`.wp-block-cover img[src$="${ testImageFilename }"]`
+		);
+
+		// Get the block's background image URL.
+		const backgroundImageURL = await backgroundImageLocator.evaluate(
+			( el ) => el.src
+		);
+		expect( backgroundImageURL ).toContain( testImageFilename );
+	} );
+
+	test( 'dims background image down by 50% by default', async ( {
+		page,
+		editor,
+	} ) => {
+		await editor.insertBlock( { name: 'core/cover' } );
+
+		// Create a test image.
+		const testImage = await getTestImage();
+
+		// Create the block using uploaded image.
+		await page
+			.locator( `.wp-block-cover input[type="file"]` )
+			.setInputFiles( testImage );
+
+		const [ backgroundDimColor, backgroundDimOpacity ] = await page
+			.locator( '.wp-block-cover .has-background-dim' )
+			.evaluate( ( el ) => {
+				const computedStyle = window.getComputedStyle( el );
+				return [ computedStyle.backgroundColor, computedStyle.opacity ];
+			} );
+
+		expect( backgroundDimColor ).toBe( 'rgb(0, 0, 0)' );
+		expect( backgroundDimOpacity ).toBe( '0.5' );
+	} );
+
+	test( 'can have the title edited', async ( { page, editor } ) => {
+		await editor.insertBlock( { name: 'core/cover' } );
+
+		// Locate the first color option from the block placeholder's color picker.
+		await page
+			.locator( 'button.components-circular-option-picker__option' )
+			.first()
+			.click();
+
+		// Click on the placeholder text to activate cursor inside.
+		const coverTitle = page.locator( '[data-title="Paragraph"]' );
+		await expect( coverTitle ).toBeVisible();
+		await coverTitle.click();
+
+		// Type the title.
+		await page.keyboard.type( 'foo' );
+		const coverTitleText = await coverTitle.innerText();
+
+		expect( coverTitleText ).toEqual( 'foo' );
+	} );
+
+	test( 'can be resized using drag & drop', async ( { page, editor } ) => {
+		await editor.insertBlock( { name: 'core/cover' } );
+
+		// Open the document sidebar.
+		await editor.openDocumentSettingsSidebar();
+
+		// Locate the first color option from the block placeholder's color picker.
+		await page
+			.locator( 'button.components-circular-option-picker__option' )
+			.first()
+			.click();
+
+		// Open the block list viewer and select child paragraph within the cover block.
+		await page.locator( '[aria-label="List View"]' ).click();
```

Comment:
**question**: I've chosen to use `aria-label` attributes instead of CSS selectors on several occasions in this file, building on past experience and the [Best Practices](https://playwright.dev/docs/selectors#best-practices) guideline from the Playwright maintainers. 

However, I see that Gutenberg repo tends to use CSS selectors under most scenarios. Is this an official guideline?

Reply:
We have our own guidelines and [best practices](https://github.com/WordPress/gutenberg/blob/trunk/test/e2e/README.md#use-accessible-selectors). The general recommendation is to use accessible selectors whenever possible. It should be really rare to resort to CSS selectors. There are several places still using CSS selectors, but they can be considered legacy code. In this case, I think a role selector will work best!

Reply:
@kevin940726 Thanks for the heads up. I think the code style was lost on me during the division meetup; I will look at updating this PR and re-request a review once done.